tests/system/src/test/java/test/javafx/accessibility/virtualflow/VirtualFlowMemoryLeakTest.java

```diff
@@ -0,0 +1,192 @@
+import javafx.stage.Stage;
+import test.util.Util;
+
+public class VirtualFlowMemoryLeakTest {
+
+    private static CountDownLatch startupLatch;
+    private static CountDownLatch screenReaderLatch = new CountDownLatch(1);
+
+    public static class TestApp extends Application {
+        @Override
+        public void start(final Stage stage) throws Exception {
+            final Scene scene = new Scene(createRoot(), 200, 200);
+            stage.setScene(scene);
+            Platform.runLater(startupLatch::countDown);
+            stage.show();
+
+            Platform.accessibilityActiveProperty()
+                    .addListener((obs, prevActive, active) -> updateScreenReaderStatus(active));
+            updateScreenReaderStatus(Platform.isAccessibilityActive());
+        }
+    }
+
+    private static void updateScreenReaderStatus(final boolean active) {
+        if (active) {
+            screenReaderLatch.countDown();
+        }
+    }
+
+    private final static ObservableList<Item> items = FXCollections.observableArrayList();
+    private final static AtomicLong counter = new AtomicLong();
+
+    static class Item {
+
+        private final ReadOnlyObjectProperty<Long> id;
+
+        public Item(final long id) {
+            this.id = new SimpleObjectProperty<>(id);
+        }
+
+        public ReadOnlyObjectProperty<Long> idProperty() {
+            return id;
+        }
+    }
+
+    private static Parent createRoot() {
+        tableView = createTableView();
+        return tableView;
+    }
+
+    private static void addItem() {
+        items.add(0, new Item(counter.incrementAndGet()));
+
+        final TableViewSelectionModel<Item> sm = tableView.getSelectionModel();
+        if (sm.getSelectedItems().isEmpty()) {
+            sm.selectLast();
+        }
+        if (!tableView.isFocused()) {
+            tableView.requestFocus();
+        }
+    }
+
+    private static final AtomicBoolean itemsPopulatedAndCleared = new AtomicBoolean(false);
+    private static WeakReference<TableRow<Item>> firstRowRef;
+
+    private static TableView<Item> tableView;
+
+    private static TableView<Item> createTableView() {
+        final TableView<Item> tableView = new TableView<>(items);
+
+        tableView.setRowFactory(param -> {
+            final TableRow<Item> row = new TableRow<>();
+            if (itemsPopulatedAndCleared.get()) {
+                if (firstRowRef == null) {
+                    firstRowRef = new WeakReference<>(row);
+                }
+            }
+            return row;
+        });
+
+        final TableColumn<Item, Long> idColumn = new TableColumn<>();
+        idColumn.setCellValueFactory(cd -> cd.getValue().idProperty());
+
+        tableView.getColumns().add(idColumn);
+        return tableView;
+    }
+
+    @BeforeClass
+    public static void initFX() {
+        startupLatch = new CountDownLatch(1);
+        new Thread(() -> Application.launch(TestApp.class, (String[]) null)).start();
+        try {
```

Comment:
It is better to replace the bunch of if .. try .. catch with assertions.

```
public static void initFX() throws InterruptedException {
    ..
    ..
    assertTrue("Timeout waiting for FX runtime to start", startupLatch.await(10, SECONDS));
}
```

Reply:
Yes, that does seem cleaner, but I note that we have several other tests that do it this way (and I'm sure this logic was copied from one of them), so I would recommend to address this as part of a larger cleanup to unify the startup of system tests that need to launch an FX Application (e.g., some of newer tests wait for a Stage shown event).

Reply:
I note that as part of the review for [JDK-8199614](https://bugs.openjdk.java.net/browse/JDK-8199614) I suggested to file a similar follow-up issue to unify the startup for tests, but it looks like the bug didn't get filed. I just filed [JDK-8206430](https://bugs.openjdk.java.net/browse/JDK-8206430) to track this.